src/agreement.rs

```diff
@@ -210,15 +210,17 @@ impl<NodeUid: Clone + Eq + Hash> Agreement<NodeUid> {
     /// can, however, expect every good node to send an AUX value that will
     /// eventually end up in our bin_values.
     fn count_aux(&self) -> (usize, BTreeSet<bool>) {
-        let mut vals: BTreeSet<bool> = BTreeSet::new();
-
-        for b in self.received_aux.values() {
-            if self.bin_values.contains(b) {
-                vals.insert(b.clone());
-            }
-        }
+        let mut count = 0;
+        let vals: BTreeSet<bool> = self.received_aux
+            .values()
+            .filter(|b| {
+                count += 1;
```

Comment:
Now it will count the total number of entries in `received_aux`, instead of just the ones whose values are in `bin_values`.